src/basicstruct.py

```diff
@@ -62,6 +77,14 @@ def __ne__(self, other):
     def __hash__(self):
         return hash(self._to_tuple())
 
+    def __iter__(self):
+        """Yield pairs of (attrubute_name, value).
+
+        This allows using `dict(my_struct)`.
+
+        """
+        return izip(self.__slots__, self._to_tuple())
```

Comment:
No, not `izip`. Notice that there's already an import of `zip` from `six`. It makes sure to use the correct, efficient `zip` for each Python version (essentially it import `izip` in Python 2.x and `zip` in Python 3.x).